arch/arm/core/aarch64/isr_wrapper.S

```diff
@@ -88,10 +88,24 @@ SECTION_FUNC(TEXT, _isr_wrapper)
 	cmp	x1, #0
 	bne	exit
 
-	/* Check if we need to context switch */
-	ldr	x1, [x0, #_kernel_offset_to_current]
-	ldr	x2, [x0, #_kernel_offset_to_ready_q_cache]
-	cmp	x1, x2
+	/*
+	 * z_arch_get_next_switch_handle() is returning:
+	 *
+	 * - The next thread to schedule in x0
+	 * - The current thread in x1. This value is returned using the
+	 *   **old_thread parameter, so we need to make space on the stack for
+	 *   that.
+	 */
+	stp	x1, xzr, [sp, #-16]!
```

Comment:
The store pair is only to adjust `SP`  they why not just adjust Or any other trick you have here.

Reply:
the `stp` was to keep the `SP` aligned to the 16-byte boundary. Not sure what you are suggesting here.

Reply:
I mean why not just do `sub sp ,sp ,..`  instead of pushing dummy data.

Reply:
Is that any real advantage? The result is exactly the same and I honestly prefer having an `stp` always before the corresponding `ldp`.

Reply:
I think it has a bit readability issue. One would think you are passing some argument to the API where as the purpose is to just create space for out param.  And as a bonus you save some cycles !

Reply:
Well, for the sake of readability I would like to force my hand here and leave this as is (as long as another reviewer steps in to give his opinion on the matter :) to keep balance between `ldp` and `stp`.

Reply:
While coding asm I follow compiler as golden rule.  What if you had to receive a bigger structure. 
would you do  `stp stp stp .. foo(sp) .. ldp ldp ldp ..`  in order to match ldp stp. but no strong opinion on this
+1  with /without this change.